src/gui/editors/BBEditor.cpp

```diff
@@ -264,3 +261,27 @@ void BBTrackContainerView::updatePosition()
 	//realignTracks();
 	emit positionChanged( m_currentPosition );
 }
+
+
+
+
+void BBTrackContainerView::makeSteps( bool clone )
+{
+	TrackContainer::TrackList tl = model()->tracks();
+
+	for( TrackContainer::TrackList::iterator it = tl.begin();
+		it != tl.end(); ++it )
+	{
+		if( ( *it )->type() == Track::InstrumentTrack )
+		{
+			Pattern* p = static_cast<Pattern *>( ( *it )->getTCO( m_bbtc->currentBB() ) );
+			if( clone )
```

Comment:
I feel that the ````if```` statement below is a better solution, than duplicating this function without the ````if```` and hard coding ````p->cloneSteps();````  in one and ````p->addSteps();```` in the other.